Rename archives/teams -> orgs in codebase + add db migration

tw4l opened the pull request:
Connected to #458 

This PR modifies the API and internal naming for both the backend and frontend to move from Archives and aids to Orgs and oids. It's a larger scope that simply renaming the API but hopefully taking the extra time will prevent any future confusion re: legacy names in the codebase.

Because some of the changes to the database collections are breaking changes, this PR also introduces database migrations (up-only for now), run after `db_init()` is called in `btrixcloud.main`, and a first migration module. If future migrations are added in the future we can use the `Migration` class as a template for those, or separate it out and have each migration module subclass it. For now since there's only the one, it's a pretty straightforward implementation.

To prevent migrations from running concurrently in several workers, the PR also introduces a new `btrixcloud.worker` module with a `by_one_worker` decorator that can be used to ensure that asyncio tasks run only in a single worker. This is then applied for database migrations as well as default org creation, superuser creation, and index initialization.

For convenience, I've also added a `.pre-commit-config.yaml` file which can be used to run `black`, `pylint`, and `husky` (for frontend linting via `eslint` and `prettier`) as pre-commit hooks. To install: ensure pre-commit and the linting tools are installed and on your path and then run `pre-commit install` in the project root directory.

tw4l committed:
Rename archives to orgs and aid to oid on backend

tw4l committed:
Rename archive to org and aid to oid in frontend

tw4l commented on frontend/src/pages/org/index.ts:
```diff
@@ -256,15 +256,15 @@ export class Archive extends LiteElement {
           <div class="flex font-medium" role="row">
             <div class="w-1/2 px-3 py-2" role="columnheader" aria-sort="none">
-              ${msg("Name", { desc: "Team member's name" })}
+              ${msg("Name", { desc: "Team member's name", id: "teammembername" })}
```
@SuaYoo I added a manual id here for this string because `lit-localize` was complaining about having multiple "Name" strings with different descriptions

ikreymer replied:
Relatedly, should we also rename this to 'orgmembername' and 'Organization Member's name'?
Or in a separate pass?

tw4l replied:
Done!

tw4l committed:
Remove translation artifact

tw4l committed:
Rename team -> organization

tw4l committed:
Add database migrations and run once on startup

This commit also applies the new by_one_worker decorator to other
asyncio tasks to prevent heavy tasks from being run in each worker.

tw4l committed:
Run black, pylint, and husky via pre-commit

tw4l commented:
@ikreymer @SuaYoo this is now ready for review - lmk about the timing re: other frontend changes in progress, happy to rebase as needed

SuaYoo reviewed (commented):
Frontend changes look good, thanks for handling that!

ikreymer commented on backend/btrixcloud/migrations/migration_0001_archives_to_orgs.py:
```diff
@@ -0,0 +1,107 @@
+    def __init__(self, mdb):
+        self.mdb = mdb
+
+    async def migrate_up_needed(self):
```
I wonder if we should add a db version information somewhere, to make it clearer when migration is needed, and to which version?

tw4l committed:
Set db version and use in migrations

tw4l committed:
Update and prepare database in single task

SuaYoo commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,22 @@
+      language: system
+      entry: frontend/.husky/pre-commit
+      pass_filenames: false
+      always_run: true
```
Curious why `always_run` for frontend?

tw4l replied:
Good point! I copied an example and forgot to remove that - now removed :)

tw4l committed:
Migrate k8s configmaps

tw4l committed:
Remove always_run from husky hook

tw4l committed:
Don't manually patch managedfields
